### PR TITLE
[16.0][MIG]dms_attachment_link: port to 16.0 version

### PR DESCRIPTION
**Pull request by DemchukM**
Migrate dms_attachment_link module from Odoo 15.0 to Odoo 16.0

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Thanks for the contribution.

Please preserve commit history following technical method explained in https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0.

If the jump is between several versions, you have to modify the source branch in the main command to accommodate it to this circumstance.

**Comment by Aldeigja** [comment]
> Please preserve commit history following technical method explained in https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0.

@pedrobaeza we have updated PR with the commit history preserved.
Please check if this is good enough now for merging.
Thanks.

**Review by angelinaanaki — APPROVED** [review]
Code LGTM

**Comment by pedrobaeza** [comment]
Formally seems OK, now reviewers should check the migration itself.

/ocabot migration dms_attachment_link

File: dms_attachment_link/models/ir_attachment.py [hunk 1]
```diff
@@ -17,3 +17,12 @@ def _compute_datas(self):
...
         return res
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        """Create attachments and link them to DMS files."""
+        attachments = super().create(vals_list)
+        for attachment, vals in zip(attachments, vals_list):
+            if "dms_file_id" in vals:
+                attachment.datas = attachment.dms_file_id.content
+        return attachments
```
Comment (victoralmau) [inline dms_attachment_link/models/ir_attachment.py]: Why is this change necessary? The `_compute_datas()` method already takes care of it (or should).

**Review by victoralmau — CHANGES_REQUESTED** [review]

File: dms_attachment_link/models/ir_attachment.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+
+class IrAttachment(models.Model):
+    _inherit = "ir.attachment"
+
+    dms_file_id = fields.Many2one(comodel_name="dms.file")
+
+    @api.depends("dms_file_id.content")
+    def _compute_datas(self):
+        """Get the contents of the attachment directly from the DMS fiel."""
```
Comment (hoangtrann) [inline dms_attachment_link/models/ir_attachment.py]:
```suggestion
        """Get the contents of the attachment directly from the DMS file."""
```

**Review by hoangtrann — APPROVED** [review]
In general code LGTM, agree with @victoralmau that extending `_compute_datas()` isn't neccessary

**Comment by mostafabarmshory** [comment]
Dear all, Would you fix and merge with the main 16.0 branch. 
tx

**Comment by mostafabarmshory** [comment]
@DemchukM 

Seams unit testes fail. This prevent the merge process?!

**Review by victoralmau — CHANGES_REQUESTED** [review]
Code review OK.

The changes from commit _[16.0][FIX]dms_attachment_link: refactoring code_ add them to _[16.0][MIG]dms_attachment_link: port to 16.0 version_

The errors in tests will already be solved when you run it again (it was related to https://github.com/OCA/oca-ci/pull/78).

**Comment by mostafabarmshory** [comment]
The branch 17.0 dose not contain this module. What the problem is?

**Comment by victoralmau** [comment]
> The branch 17.0 dose not contain this module. What the problem is?

The 17.0 branch does not have this module because it has not been migrated. If you want to migrate it you can do it (for example when this migration is finished and it is merged in v16).

**Comment by victoralmau** [comment]
Superseed by https://github.com/OCA/dms/pull/371

**Pull request closed without merging**